### ISSUE TITLE
Incorrect processing of Outline examples

### ISSUE THREAD

**Issue by ptrimble** [sittercity/gocumber#15]
I am not sure I am understanding how the outline examples are processed. Based on my understanding we are not processing them correctly. Let me provide an example. For this gherkin:

```gherkin
Feature: Scenario Outline with DocString

Scenario Outline: Something
  Given something is:
    """
    <how> functional <really_though>
    """

Examples: 
  | how         | really_though    |
  | minimally | barely                |
  | incredibly | overwhelmingly |
```

I imagine we should run this twice, once for each example in the table. So we run `Given something is` step twice in total.

Based on my reading of how we process it today we only run it once. Does this line only go through the steps in the outline once? Shouldn't we be looping over the examples, not the steps? Here is the line: https://github.com/sittercity/gocumber/blob/master/gocumber.go#L156

I wrote a test like this:

```go
func TestDocstrings_OutlineVariables(t *testing.T) {
        steps := make(Definitions)
        tt := new(testing.T)

        var called bool
        steps.Given("something is:", func(_ []string, step StepNode) {
                called = true
                assert.Equal(t, "minimally functional\n\n", step.PyString().String())
        })

        steps.Run(t, "test/valid_with_pystring_outline.feature")

        assert.True(t, called)
        assert.False(t, tt.Failed())
}
```

With the gherkin above. I put in a print statement like so:

```go
        for _, original := range outline.Steps() {
                fmt.Printf("first outline: %v\n", original.Text())
                step := nodes.NewMutableStepNode(original.StepType(), replace(original.Text()))

                if original.Table() != nil {
                        table := nodes.NewMutableTableNode()

                        for _, row := range original.Table().Rows() {
                                replaced := make([]string, len(row))

                                for i, cell := range row {
                                        replaced[i] = replace(cell)
                                }

                                table.AddRow(replaced)
                        }

                        step.SetTable(table)
                } else if original.PyString() != nil {
                        pyString := nodes.NewMutablePyStringNode()
                        pyString.AddLine(replace(original.PyString().String()))

                        step.SetPyString(pyString)
                }

                callback(step)
        }
```

This is my output:

```sh
% make test
godep go test -cover .
first outline: something is:
--- FAIL: TestDocstrings_OutlineVariablesMultipleExamples (0.00s)
        gocumber.go:168: Scenario: Something
        Location:       gocumber_test.go:228
        Error:          Not equal: "minimally functional barely\n\n" (expected)
                                != "incredibly functional overwhelmingly\n\n" (actual)

FAIL
coverage: 100.0% of statements
FAIL    github.com/sittercity/gocumber  0.019s
godep: go exit status 1
make: *** [unit-test] Error 1
```

It seems to me that we should loop over the examples so we have two steps for this scenario outline.

**Comment by cbandy** [comment]
Yep, the function you linked should loop over the steps from inside a loop over the examples.

It looks like moving [this brace](https://github.com/sittercity/gocumber/blob/master/gocumber.go#L154) to the end of the function should do.

**Comment by ptrimble** [comment]
Cool, that's exactly what I am doing now. I'll have a PR shortly.

**Comment by ptrimble** [comment]
Addressed in PR #17
